packages/server/src/routes/grants.js

```diff
@@ -431,15 +431,33 @@ router.put('/:grantId/interested/:agencyId', requireUser, async (req, res) => {
         return;
     }
 
-    await db.markGrantAsInterested({
-        grantId,
-        agencyId,
-        userId: user.id,
-        interestedCode,
-    });
-
-    const interestedAgencies = await db.getInterestedAgencies({ grantIds: [grantId], tenantId: user.tenant_id });
-    res.json(interestedAgencies);
+    try {
+        // Calling the existing function to mark the grant as interested
+        await db.markGrantAsInterested({
+            grantId,
+            agencyId,
+            userId: user.id,
+            interestedCode,
+        });
+        // Query to check if the interestedCode corresponds to 'Interested'
+        const interestedStatus = await knex('interested_codes')
+            .select('status_code')
+            .where({ id: interestedCode })
+            .first();
+
+        // Follow  or Unfollow the grant based on the interestedCode
+        if (interestedStatus?.status_code === 'Interested') {
+            await followGrant(knex, grantId, user.id);
+        } else {
+            await unfollowGrant(knex, grantId, user.id);
+        }
+        // Retruning updated interested agencies
+        const interestedAgencies = await db.getInterestedAgencies({ grantIds: [grantId], tenantId: user.tenant_id });
+        res.json(interestedAgencies);
+    } catch (error) {
+        console.error('Error in marking as interested:', error);
+        res.status(500).send({ success: false, error: 'Internal Server Error' });
+    }
```

Comment:
I'd prefer to remove this wrapping try/catch block since we're not inspecting it or otherwise using it to provide any additional context to the user if something were to fail. Otherwise the try/catch will suppress the error and make this harder to monitor and debug (in Datadog).